pkg/filter/filter.go

```diff
@@ -108,18 +112,36 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
 // Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, obj runtime.Object) bool {
```

Comment:
I know we said to use runtime.Object, but I wonder if metav1.Object would be easier to use since it has a `GetLabels()` method

Reply:
Should the input here just be `labels map[string]string`? The caller can extract the labels and pass those in

Reply:
@vkamra that may be a better approach given that we can create a LabelSelector and then Selector and pass the map to the Selector's Matches method. This would make the use of the filter pkg simpler.

Reply:
Have moved to map as @vkamra suggested. This may require a change in how clients create a ResourceMatcher in order to get labels to be used. I also need to add tests to cover the changes before making this ready for review.